.github/workflows/workflow.yml

```diff
@@ -45,6 +42,23 @@ jobs:
       run: tox -e copyright-check
     - name: Manifest check
       run: tox -e check-manifest
+
+  unit_tests:
+    needs:
+    - code_quality_check
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.7, 3.8, 3.9, "3.10"]
+    timeout-minutes: 30
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: pip install tox==3.25.1
     - name: Tests and coverage
       run: |
         tox -e test-unit
```

Comment:
Change dependencies to only tests?

Reply:
Maybe we should give poetry another try? I remember you saying it has these features. Should experiment with the performance concerns which were the main reason we didn't follow through and if still there (which wasn't a large diff with pipenv if I remember correctly) we may want to consider switching anyway for cosmpy just because the deps aren't very big (in comparison with something like the AEA framework).

Reply:
decided to check on a dedicated PR